src/main/java/com/wultra/security/userdatastore/controller/DocumentController.java

```diff
@@ -0,0 +1,134 @@
+/*
+ * User Data Store
+ * Copyright (C) 2024 Wultra s.r.o.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as published
+ * by the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+package com.wultra.security.userdatastore.controller;
+
+import com.wultra.security.userdatastore.model.dto.DocumentDto;
+import com.wultra.security.userdatastore.model.request.DocumentCreateRequest;
+import com.wultra.security.userdatastore.model.request.DocumentUpdateRequest;
+import com.wultra.security.userdatastore.model.response.DocumentCreateResponse;
+import com.wultra.security.userdatastore.model.response.DocumentResponse;
+import com.wultra.security.userdatastore.service.DocumentService;
+import io.getlime.core.rest.model.base.response.ObjectResponse;
+import io.getlime.core.rest.model.base.response.Response;
+import io.swagger.v3.oas.annotations.Operation;
+import jakarta.validation.constraints.NotBlank;
+import jakarta.validation.constraints.Size;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.validation.annotation.Validated;
+import org.springframework.web.bind.annotation.*;
+
+import java.util.List;
+
+/**
+ * REST controller providing API for CRUD for user documents.
+ *
+ * @author Roman Strobl, roman.strobl@wultra.com
+ */
+@RestController
+@Validated
+@Slf4j
+class DocumentController {
+
+    private final DocumentService documentService;
+
+    DocumentController(DocumentService documentService) {
+        this.documentService = documentService;
+    }
+
+
+    /**
+     * Return documents for the given user.
+     *
+     * @param userId user identifier
+     * @param documentId optional document identifier
+     * @return user documents
+     */
+    @Operation(
+            summary = "Return documents",
+            description = "Return documents for the given user."
+    )
+    @GetMapping("/documents")
+    public ObjectResponse<DocumentResponse> userDocuments(@NotBlank @Size(max = 255) @RequestParam String userId, @Size(max = 255) @RequestParam(required = false) String documentId) {
+        if (documentId == null) {
```

Comment:
I suggest to use a single log message `logger.info("Fetching documents; user ID: {}, document ID: {}", userId, documentId);`
The branch could be logged in the service.